### ISSUE TITLE
Transforming objects created by `Object.create(null)` produce unexpected results.

### ISSUE THREAD

**Issue by blaugold** [typestack/class-transformer#120]
When transforming a plain object created by `Object.create(null)` `plainToClass` does not return a new instance of the given constructor. Instead, it returns the original object:
```typescript
import { plainToClass } from 'class-transformer'

class A {

  a: string

}

function initObject(o: any): any {
  o.a = 'A'
  return o
}


const objectWithoutPrototype = Object.create(null)
const object                 = {}

const instanceFromObject                 = plainToClass(A, initObject(object))
const instanceFromObjectWithoutPrototype = plainToClass(A, initObject(objectWithoutPrototype))

printInstance('instanceFromObject', instanceFromObject)
printInstance('instanceFromObjectWithoutPrototype', instanceFromObjectWithoutPrototype)

console.log('instanceFromObjectWithoutPrototype === objectWithoutPrototype',
  instanceFromObjectWithoutPrototype === objectWithoutPrototype)

function printInstance(name: string, inst: any) {
  const prototype   = Object.getPrototypeOf(inst)
  const constructor = prototype && prototype.constructor

  console.log(`${name}:`)
  console.log('prototype', prototype)
  console.log('constructor', constructor && constructor.name)
  console.log('instance', inst)
  console.log('')
}
```

When exectued:
```
instanceFromObject:
prototype A {}
constructor A
instance A { a: 'A' }

instanceFromObjectWithoutPrototype:
prototype null
constructor null
instance { a: 'A' }

instanceFromObjectWithoutPrototype === objectWithoutPrototype true
```

"class-transformer": "^0.1.8"

**Comment by bjyoungblood** [comment]
:+1: here. Apparently the Express.js team decided that it's best for user-supplied objects to have a null prototype (see links below). Unfortunately, this means that `instanceof Object` checks will fail on these objects.

https://github.com/expressjs/multer/pull/187
https://github.com/expressjs/discussions/issues/25
https://github.com/expressjs/multer/pull/318

The culprit is [here](https://github.com/typestack/class-transformer/blob/66a0dcc23b4b4dfd13b6b380c542b7afda7c90d9/src/TransformOperationExecutor.ts#L77).

The simplest fix would be to add `|| typeof value === 'object'` to the condition.

As for the potential for future debugging, my personal preference would be to throw an error from the else case of `TransformOperationExecutor::transform` instead of returning the original value, especially since returning the original value violates the return type contract defined by `plainToClass`.

**Comment by blaugold** [comment]
I stumbled across this issue because GraphQL does the same.

**Comment by NoNameProvided** [comment]
Thanks for the heads up @bjyoungblood! 

This looks serious. Can you propose a PR with your minimal fix for this?

**Comment by NoNameProvided** [comment]
This has been fixed in 0cd39ed and will be included in the next release.

**Comment by NoNameProvided** [comment]
Just published `0.1.9`, please test it and report if you still encounter this issue. Thanks!
